python/lsst/ctrl/mpexec/cli/script/qgraph.py

```diff
@@ -0,0 +1,156 @@
+        if `extended-run` flag is set.
+    save_qgraph : `str` or `None`
+        Path location for storing a serialized quantum graph definition as a
+        pickle file.
+    save_single_quanta : `str` or `None`
+        Format string of locations for storing individual quantum graph
+        definition (pickle files). The curly brace {} in the input string will
+        be replaced by a quantum number.
+    qgraph_dot : `str` or `None`
+        Path location for storing GraphViz DOT representation of a quantum graph.
+    butler_config : `str`, `dict`, or `lsst.daf.butler.Config`
+        If `str`, `butler_config` is the path location of the gen3
+        butler/registry config file. If `dict`, `butler_config` is key value
+        pairs used to init or update the `lsst.daf.butler.Config` instance. If
+        `Config`, it is the object used to configure a Butler.
+    input : `str`
+        Comma-separated names of the input collection(s). Entries may include a
+        colon (:), the first string is a dataset type name that restricts the
+        search in that collection.
+    output : `str`
+        Name of the output CHAINED collection. This may either be an existing
+        CHAINED collection to use as both input and output (if `input` is
+        `None`), or a new CHAINED collection created to include all inputs
+        (if `input` is not `None`). In both cases, the collection's children
+        will start with an output RUN collection that directly holds all new
+        datasets (see `output_run`).
+    output_run : `str`
+        Name of the new output RUN collection. If not provided then `output`
+        must be provided and a new RUN collection will be created by appending
+        a timestamp to the value passed with `output`. If this collection
+        already exists then `extend_run` must be passed.
+    extend_run : `bool`
+        Instead of creating a new RUN collection, insert datasets into either
+        the one given by `output_run` (if provided) or the first child
+        collection of `output` (which must be of type RUN).
+    replace_run : `bool`
+        Before creating a new RUN collection in an existing CHAINED collection,
+        remove the first child collection (which must be of type RUN). This can
+        be used to repeatedly write to the same (parent) collection during
+        development, but it does not delete the datasets associated with the
+        replaced run unless `prune-replaced` is also True. Requires `output`,
+        and `extend_run` must be `None`.
+    prune_replaced : "unstore", "purge", or `None`.
+        If not `None`, delete the datasets in the collection replaced by
+        `replace_run`, either just from the datastore ("unstore") or by
+        removing them and the RUN completely ("purge"). Requires `replace_run`.
+    data_query : `str`
+        User query selection expression.
+    show : `list` [`str`] or `None`
+        Descriptions of what to dump to stdout.
```

Comment:
There is a lot of duplication between this docstring and help strings in option definitions. I'm not sure that we need to repeat them here, this is not a public API, and we want to avoid repetition for this sort of descriptions.

Reply:
I really don't know how to slice this in python. I think it's super maddening to look at a function without any documentation and have to reverse engineer the types and intents of each argument, or to know where to look up or down the stack (after having figured out what the stack will be) to find the documentation and to then know if that documentation applies here. Yes this is verbose, but at least when I'm here I know what was intended by the author or last person to make changes and IMO that's better than the guessing game.

Reply:
My concern is that description here and description of CLI arguments can get out of sync. We probably need `SuperClick` which can read method docstring and build arguments from that 🙂